packages/core/test/test-util/filter-factory.ts

```diff
@@ -109,3 +118,12 @@ export const testFilterFunctionNested = {
   odataStr:
     '(substringof(to_SingleLink/StringProperty, to_SingleLink/KeyProperty) eq false)'
 };
+
+export const testFilterLambdaExpressionOnLink = {
+  filter: TestEntityV4.TO_MULTI_LINK.filter(
+    any(TestEntityMultiLinkV4.STRING_PROPERTY.equals('test1')),
```

Comment:
Could you add a type check for the filter. I replaced the `TestEntityMultiLinkV4` with the plain `TestEntity` which I thought should lead to an error but at least my IDE did not show one.

Reply:
Could you please copy the whole expression? I tested by replacing the v4 entity with v2, it failed as expected.
In general, before merging the generator, it's not easy to create the type tests, since the dummy vdms are not there.
So the idea is to make manual tests and add automated tests here: https://github.wdf.sap.corp/MA/scp-rocket/issues/1267 once the v4 generator is available.

Reply:
@jjtang1985 The new test services are there. They only don't support function imports, but everything else is there.